File: vibebot/core/message_utils.py
from datetime import date, datetime


def build_header() -> list[dict]:
    today = date.today().strftime("%A, %B %-d, %Y")
    now = datetime.now().strftime("%-I:%M %p")
    return [
        {
            "type": "header",
            "text": {
                "type": "plain_text",
                "text": f"🤖 VibeBot Daily Digest — {today}",
                "emoji": True,
            },
        },
        {
            "type": "context",
            "elements": [
                {
                    "type": "mrkdwn",
                    "text": f"Your AI-curated morning briefing  ·  Sent at {now}",
                }
            ],
        },
    ]


def build_footer() -> list[dict]:
    return [
        {"type": "divider"},
        {
            "type": "context",
            "elements": [{"type": "mrkdwn", "text": "🤖 _Powered by VibeBot_"}],
        },
    ]

File: src/core/messageUtils.ts
export function buildHeader(): object[] {
  const now = new Date();
  const today = now.toLocaleDateString('en-US', { weekday: 'long', month: 'long', day: 'numeric', year: 'numeric' });
  const time = now.toLocaleTimeString('en-US', { hour: 'numeric', minute: '2-digit', hour12: true });

  return [
    {
      type: 'header',
      text: {
        type: 'plain_text',
        text: `🤖 VibeBot Daily Digest — ${today}`,
        emoji: true,
      },
    },
    {
      type: 'context',
      elements: [
        {
          type: 'mrkdwn',
          text: `Your AI-curated morning briefing  ·  Sent at ${time}`,
        },
      ],
    },
  ];
}

export function buildFooter(): object[] {
  return [
    { type: 'divider' },
    {
      type: 'context',
      elements: [{ type: 'mrkdwn', text: '🤖 _Powered by VibeBot_' }],
    },
  ];
}
